flash/video/classification/data.py

```diff
@@ -265,19 +140,68 @@ def from_paths(
 
         clip_sampler = make_clip_sampler(clip_sampler, clip_duration, **clip_sampler_kwargs)
 
-        preprocess: Preprocess = preprocess or cls.preprocess_cls(
-            clip_sampler, video_sampler, decode_audio, decoder, train_transform, val_transform, test_transform,
-            predict_transform
+        super().__init__(
+            train_transform=train_transform,
+            val_transform=val_transform,
+            test_transform=test_transform,
+            predict_transform=predict_transform,
+            data_sources={
+                DefaultDataSources.PATHS: VideoClassificationPathsDataSource(
+                    clip_sampler,
+                    video_sampler=video_sampler,
+                    decode_audio=decode_audio,
+                    decoder=decoder,
+                )
+            },
+            default_data_source=DefaultDataSources.PATHS,
         )
 
-        return cls.from_load_data_inputs(
-            train_load_data_input=train_data_path,
-            val_load_data_input=val_data_path,
-            test_load_data_input=test_data_path,
-            predict_load_data_input=predict_data_path,
-            batch_size=batch_size,
-            num_workers=num_workers,
-            preprocess=preprocess,
-            use_iterable_auto_dataset=True,
-            **kwargs,
-        )
+    def get_state_dict(self) -> Dict[str, Any]:
+        return {
+            'train_transform': self._train_transform,
```

Comment:
I think we should swap between `_train_transform` and `train_transform` in the preprocess internals.
The public should be the provided by the user.

Reply:
Done :smiley: